velox/functions/lib/string/StringCore.h

```diff
@@ -487,20 +505,40 @@ getByteRange(const char* str, size_t startCharPosition, size_t length) {
     size_t startByteIndex = 0;
     size_t nextCharOffset = 0;
 
+    while (nextCharOffset < strLength && utf_cont(str[nextCharOffset])) {
+      nextCharOffset++;
+    }
+
     // Find startByteIndex
-    for (auto i = 0; i < startCharPosition - 1; i++) {
-      auto increment = utf8proc_char_length(&str[nextCharOffset]);
-      nextCharOffset += UNLIKELY(increment < 0) ? 1 : increment;
+    for (auto i = 0; nextCharOffset < strLength && i < startCharPosition - 1;
+         i++) {
+      nextCharOffset++;
+
+      while (nextCharOffset < strLength && utf_cont(str[nextCharOffset])) {
```

Comment:
I like the idea of using a lambda since I do it in a few places, plus it makes it easy to comment on what I'm doing since it centralizes it in a single place.